netbox/dcim/views.py

```diff
@@ -122,14 +122,18 @@ def post(self, request):
             if form.is_valid():
 
                 with transaction.atomic():
-
+                    cables = []
                     count = 0
                     for obj in self.queryset.filter(pk__in=form.cleaned_data['pk']):
                         if obj.cable is None:
                             continue
-                        obj.cable.delete()
+                        if obj.cable.pk not in cables:
+                            cables.append(obj.cable.pk)
                         count += 1
 
+                    if len(cables) > 0:
+                        Cable.objects.filter(pk__in=cables).delete()
```

Comment:
Are we sure that calling `delete()` on the queryset (as opposed to each Cable object individually) is safe?

Reply:
You are correct, it might be better to call each cable as it skips the delete() method and does a bulk delete anyways.

```suggestion
                    for cable in Cable.objects.filter(pk__in=cables):
                        cable.delete()
```

Do you think this would be a reasonable change instead?